src/configuration.ts

```diff
@@ -13,164 +13,159 @@ import { PathResolver } from "./resolvers/path-resolver";
-            debug: this.debug,
-        };
-        this.logger.logInfo('CONFIGURATION', settings);
-        return settings;
+    const resourcesSettings: Array<ResourceSettings> = [];
+
+    for (let index = 0; index < workspace.workspaceFolders.length; index++) {
+      const resource = workspace.workspaceFolders[index].uri;
+      const config = workspace.getConfiguration("phpsab", resource);
+      const rootPath = this.resolveRootPath(resource);
+      let settings: ResourceSettings = {
+        fixerEnable: config.get("fixerEnable", true),
+        fixerArguments: config.get("fixerArguments", []),
+        workspaceRoot: rootPath,
+        executablePathCBF: config.get("executablePathCBF", ""),
+        executablePathCS: config.get("executablePathCS", ""),
+        composerJsonPath: config.get("composerJsonPath", "composer.json"),
+        standard: config.get("standard", ""),
+        autoRulesetSearch: config.get("autoRulesetSearch", true),
+        allowedAutoRulesets: config.get("allowedAutoRulesets", [
+          ".phpcs.xml",
+          "phpcs.xml",
+          "phpcs.dist.xml",
+          "ruleset.xml",
+        ]),
+        snifferEnable: config.get("snifferEnable", true),
+        snifferArguments: config.get("snifferArguments", []),
+      };
+
+      settings = await this.resolveCBFExecutablePath(settings);
+      settings = await this.resolveCSExecutablePath(settings);
+
+      settings = await this.validate(
+        settings,
+        workspace.workspaceFolders[index].name
+      );
+
+      resourcesSettings.splice(index, 0, settings);
     }
 
-    /**
-     * Attempt to find the root path for a workspace or resource
-     * @param resource
-     */
-    private resolveRootPath(resource: Uri) {
-        // try to get a valid folder from resource
-        let folder = workspace.getWorkspaceFolder(resource);
-
-        // one last safety check
-        return folder ? folder.uri.fsPath : "";
+    // update settings from config
+    let settings: Settings = {
+      resources: resourcesSettings,
+      snifferMode: this.config.get("snifferMode", "onSave"),
+      snifferShowSources: this.config.get("snifferShowSources", false),
+      snifferTypeDelay: this.config.get("snifferTypeDelay", 250),
+      debug: this.debug,
+    };
+    this.logger.logInfo("CONFIGURATION", settings);
+    return settings;
+  }
+
+  /**
+   * Attempt to find the root path for a workspace or resource
+   * @param resource
+   */
+  private resolveRootPath(resource: Uri) {
+    // try to get a valid folder from resource
+    let folder = workspace.getWorkspaceFolder(resource);
+
+    // one last safety check
+    return folder ? folder.uri.fsPath : "";
+  }
+
+  /**
+   * Get correct executable path from resolver
+   * @param settings
+   */
+  protected async resolveCBFExecutablePath(
+    settings: ResourceSettings
+  ): Promise<ResourceSettings> {
+    if (settings.executablePathCBF === null || settings.executablePathCBF === "") {
```

Comment:
I think this can probably just be if (!settings.executablePathCBF)

Reply:
That's a good point, I will check.